platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/VersionCatalogDependencyRegistry.java

```diff
@@ -0,0 +1,171 @@
+        return versions.values();
+    }
+
+    public Collection<LibraryEntry> getLibraries() {
+        return libraries.values();
+    }
+
+    public Collection<PluginEntry> getPlugins() {
+        return plugins.values();
+    }
+
+    public String registerLibrary(String module, String version) {
+        String alias = fullyQualifiedAliases ? moduleToAlias(module) : artifactToAlias(module);
+        VersionEntry versionEntry = findOrCreateVersionEntry(alias, module, version);
+        LibraryEntry libraryEntry = findOrCreateLibraryEntry(alias, module, versionEntry);
+        return "libs." + libraryEntry.alias.replaceAll("-", ".");
+    }
+
+    public String registerPlugin(String pluginId, String version) {
+        String alias = fullyQualifiedAliases ? moduleToAlias(pluginId) : pluginIdToAlias(pluginId);
+        PluginEntry pluginEntry = findOrCreatePluginEntry(alias, pluginId, version);
+        return "libs.plugins." + pluginEntry.alias.replaceAll("-", ".");
+    }
+
+    private VersionEntry findOrCreateVersionEntry(String alias, String module, String version) {
+        for (VersionEntry v : versions.values()) {
+            if (v.module.equals(module) && v.version.equals(version)) {
+                return v;
+            }
+        }
+        VersionEntry v = new VersionEntry();
+        v.alias = findFreeAlias(versions.keySet(), alias);
+        v.module = module;
+        v.version = version;
+        versions.put(v.alias, v);
+        return v;
+    }
+
+    private LibraryEntry findOrCreateLibraryEntry(String alias, String module, VersionEntry versionEntry) {
+        for (LibraryEntry l : libraries.values()) {
+            if (l.module.equals(module) && l.version.equals(versionEntry.version)) {
+                return l;
+            }
+        }
+        LibraryEntry l = new LibraryEntry();
+        if (RESERVED_LIBRARY_PREFIX.matcher(alias).find()) {
+            alias = "my" + alias;
+        }
+        l.alias = findFreeAlias(libraries.keySet(), alias);
+        l.module = module;
+        l.version = versionEntry.version;
+        l.versionRef = versionEntry.alias;
+        libraries.put(l.alias, l);
+        return l;
+    }
+
+    private PluginEntry findOrCreatePluginEntry(String alias, String pluginId, String version) {
+        for (PluginEntry p : plugins.values()) {
+            if (p.pluginId.equals(pluginId) && p.version.equals(version)) {
+                return p;
+            }
+        }
+        PluginEntry p = new PluginEntry();
+        p.alias = findFreeAlias(plugins.keySet(), alias);
+        p.pluginId = pluginId;
+        p.version = version;
+        plugins.put(p.alias, p);
+        return p;
+    }
+
+    private static String pluginIdToAlias(String pluginId) {
+        String[] pluginIdComponents = pluginId.split("\\.");
+        return moduleToAlias(pluginIdComponents[pluginIdComponents.length - 1]);
+    }
+
+    private static String artifactToAlias(String module) {
```

Comment:
💅 Using "Artifact" here doesn't make sense as per my other comment above.

This method should be named `moduleToAlias()` and the one below `coordinatesToAlias()`.